src/Compilers/CSharp/Test/Semantic/Semantics/SpanStackSafetyTests.cs

```diff
@@ -1440,5 +1440,38 @@ static void Main()
                 Diagnostic(ErrorCode.ERR_BadSKunknown, "S1").WithArguments("Program.S1", "type").WithLocation(16, 24)
                 );
         }
+
+        [Fact]
+        [WorkItem(27874, "https://github.com/dotnet/roslyn/issues/27874")]
+        public void PassingSpansToLocals_EscapeScope()
+        {
+            CompileAndVerify(
+                CreateCompilationWithMscorlibAndSpan(@"
+using System;
+class C
+{
+    static void Main()
+    {
+        Span<int> x = stackalloc int [10];
+        
+        Console.WriteLine(M1(ref x).Length);
+        Console.WriteLine(M2(ref x).Length);
+    }
+    
+    static ref Span<int> M1(ref Span<int> x)
+    {
+        ref Span<int> q = ref x;
```

Comment:
Can you also add a test for doing this inside a ref struct with the assignment value being a field?

Reply:
@agocke going through the rules, I'm not sure I understand your scenario. What are you trying to test here?

---
In reply to: [196612679](https://github.com/dotnet/roslyn/pull/27882#discussion_r196612679) [](ancestors = 196612679)

Reply:
When `x` is a field though it's always escapable as it has heap lifetime. Or if the container is a struct it's never escapable. I believe we already have tests for those scenarios.

Reply:
If we already have those tests, that's fine.

Reply:
@agocke please check `CodeGenRefReadOnlyReturnTests.ReadonlyReturnByRefInStruct` and `RefEscapingTests.FieldOfRefLikeEscape`

Reply:
Side note @jaredpar. I looked at those tests and was surprised that this is an error. Specifically, the error is that `this` is never ref-returnable. That's interesting because it seems like there's little difference if you see `this` as being an implicit ref parameter.

Since you also assumed it was ref escapable, maybe this is a language pitfall? Is there any safety concern with returning `this`?